### PR TITLE
Make the docker-compose env work as a dev environment

### PR DESCRIPTION
**Pull request by russell**
* Mount the app directory into the container
* Setup the env files on start
* Seed if the sqllite db file is missing

/cc @zendesk/samson

### Tasks
 - [ ] :+1: from team

### Risks
- Level: Low

### CODE REVIEW

**Comment by dasch** [comment]
@ragurney can you also review this?

File: .gitignore [hunk 1]
```diff
@@ -46,3 +46,4 @@
 # .bak files
 *.bak
 /vendor/cache/**/
+/db/development.sqlite
```
Comment (grosser) [inline .gitignore]: we already ignore all .sqlite files in that folder

File: docker-compose.yml [hunk 2]
```diff
@@ -4,8 +4,11 @@ services:
     image: zendesk/samson:latest
     ports:
       - "3000:9080"
+    volumes:
+      - .:/app/
     environment:
-      DATABASE_URL: "sqlite3://db/development.sqlite"
+      DATABASE_URL: "sqlite3:///app/db/development.sqlite"
       RAILS_LOG_TO_STDOUT: 1
+      RAILS_ENV: development
```
Comment (grosser) [inline docker-compose.yml]: not necessary

File: docker-compose.yml [hunk 3]
```diff
@@ -4,8 +4,11 @@ services:
...
     # to use .env.virtualbox with 192.168.42.45 host prefix ./bin/decode_dot_env .env.virtualbox && source <(sed -E -n 's/[^#]+/export &/ p' .env.virtualbox) &&
```
Comment (grosser) [inline docker-compose.yml]: comment should move into docker_dev_server

**Comment by grosser** [comment]
check readme/setup.md if anything needs an update ... maybe mention that it mounts the current folder / reuses sqlite database or so

File: .gitignore (hunk 1 above)
Comment (russell) [reply .gitignore]: ah, it didn't match the name in the docker-compose.yml, i've updated it.

File: docker-compose.yml (hunk 2 above)
Comment (russell) [reply docker-compose.yml]: 👍

File: docker-compose.yml (hunk 3 above)
Comment (russell) [reply docker-compose.yml]: 👍

**Comment by russell** [comment]
I added a note to the setup.md to mention the mounting

**Comment by dasch** [comment]
Merged so that I can rebase on master.

**Pull request merged**